src/packages/client/src/runtime/query.ts

```diff
@@ -1000,16 +1000,18 @@ export function transformDocument(document: Document): Document {
                 argType: ar.argType, // probably wrong but fine
                 schemaArg: ar.schemaArg // probably wrong but fine
               })])
-            } 
+            }
           }
         } else if (
           typeof ar.value === 'object'
-          && ar.schemaArg?.inputType[0].kind === 'object' 
-          && ar.key !== 'is' 
+          && ar.schemaArg?.inputType[0].kind === 'object'
+          && ar.key !== 'is'
+          && ar.argType !== 'Json'
           // do not add `is` on ...ListRelationFilter 
           // https://github.com/prisma/prisma/issues/3342
           && !(typeof ar.schemaArg?.inputType[0].type === 'object' && ar.schemaArg?.inputType[0].type.name.includes('ListRelationFilter'))
-          ) {
+          && !(typeof ar.argType === 'object' && (ar.argType as DMMF.InputType)?.name?.startsWith('Json'))
```

Comment:
What about the `NestedJsonNullableFilter`? I'm not sure if we need to do something about it 🤷‍♂️

Reply:
What is `NestedJsonNullableFilter`?

Reply:
I do not know exactly but it's in the DMMF